overlay.go

```diff
@@ -264,7 +264,8 @@ func (o *Overlay) requestTree(si *network.ServerIdentity, onetMsg *ProtocolMsg,
 		return err
 	}
 
-	err = o.server.Send(si, msg)
+	// no need to record sentLen because Overlay uses Server's CounterIO
+	_, err = o.server.Send(si, msg)
```

Comment:
Couldn't there be more than one overlay? Then the number of sent bytes in Overlay and in Server would be different. But if 1 Server has only 1 Overlay, then this is OK.

Reply:
I will think about this one.

Reply:
I don't think we have a straightforward solution for this problem. I could do something like `reflect.TypeOf(Envelope).Size()`, but that may not give us the right results because fields in a message may be pointers. Another solution is to change more of the API, i.e.
```
type Conn interface {
	...
	Receive() (*Envelope, uint64, error) // add uint64
	...
}

type Dispatcher interface {
	RegisterProcessor(p Processor, msgType ...MessageTypeID)
	RegisterProcessorFunc(MessageTypeID, func(*Envelope))
	Dispatch(*Envelope, uint64) error // add uint64
}

type Processor interface {
	Process(*Envelope, uint64) // add uint64
}
```

These changes are needed because router calls `Receive`, which would return the message size, then it calls `Dispatch` and we can pass the size down. Finally, `Dispatch` calls `Process`, which is implemented by `Overlay`, and we can record the Rx.

Reply:
How does this proposal sound @ineiti ? Another way is to just remove CounterIO from Overlay, because Crux doesn't need it (it only needs it on TreeNodeInstance).

Reply:
I'm still not sure my comment is even correct ;) Is there a way to have more than one overlay per server?

Reply:
Actually, I should've checked that. In `server.go:newServer`, it does `c.overlay = NewOverlay(c)`, so the answer is no.

Reply:
I don't `c.overlay` is replaced anywhere else.